pkg/util/validation/limits.go

```diff
@@ -360,6 +361,7 @@ func (l *Limits) RegisterFlags(f *flag.FlagSet) {
 	f.BoolVar(&l.ResultsCacheForUnalignedQueryEnabled, "query-frontend.cache-unaligned-requests", false, "Cache requests that are not step-aligned.")
 	f.IntVar(&l.MaxQueryExpressionSizeBytes, MaxQueryExpressionSizeBytesFlag, 0, "Max size of the raw query, in bytes. This limit is enforced by the query-frontend for instant, range and remote read queries. 0 to not apply a limit to the size of the query.")
 	f.BoolVar(&l.AlignQueriesWithStep, alignQueriesWithStepFlag, false, "Mutate incoming queries to align their start and end with their step to improve result caching.")
+	f.BoolVar(&l.PromQLExperimentalFunctionsEnabled, "query-frontend.promql-experimental-functions-enabled", false, "Enable experimental PromQL functions, which are subject to being changed or removed at any time.")
```

Comment:
Have you considered how this overlaps with the existing flag `-querier.promql-experimental-functions-enabled`?

Reply:
Yep in the query-frontend `querier.promql-experimental-functions-enabled` enables it in the PromQL engine and `query-frontend.promql-experimental-functions-enabled` enabled it for the tenants in the new middleware, but the matching names does make it confusing, hopefully with your update and the different name it's better now.